src/linker/Linker.Dataflow/ValueNode.cs

```diff
@@ -367,13 +368,12 @@ protected override string NodeToString ()
 	/// </summary>
 	abstract record LeafValueWithDynamicallyAccessedMemberNode : LeafValueNode
 	{
-		public LeafValueWithDynamicallyAccessedMemberNode (IMetadataTokenProvider sourceContext, DynamicallyAccessedMemberTypes dynamicallyAccessedMemberTypes)
+		public LeafValueWithDynamicallyAccessedMemberNode (DynamicallyAccessedMemberTypes dynamicallyAccessedMemberTypes)
 		{
-			SourceContext = sourceContext;
 			DynamicallyAccessedMemberTypes = dynamicallyAccessedMemberTypes;
 		}
 
-		public IMetadataTokenProvider SourceContext { get; }
+		public virtual LeafValueWithDynamicallyAccessedMemberNode SourceValue { get => this; }
```

Comment:
I think we discussed this, but I don't remember the outcome. Is this only for the `Type.AssemblyQualifiedName` case where the argument is represented as an `AnnotatedStringValue`?

Could we replace `AnnotatedStringValue` there with `MethodParameterValue` and remove `SourceValue`, or do we think this pattern (wrapping the value in a different type for tracking purposes while holding onto the source) will be useful elsewhere?